src/multipart/multipart.rs

```diff
@@ -0,0 +1,469 @@
+extern crate uuid;
+
+use std;
+use std::borrow::Cow;
+use std::io::Read;
+use std::ascii::AsciiExt;
+use hyper::mime::Mime;
+
+/// A multipart/form-data request.
+#[derive(Debug)]
+pub struct MultipartRequest {
+    boundary: String,
+    fields: Vec<MultipartField>,
+}
+
+impl MultipartRequest {
+    /// Creates a new MultipartRequest without any content.
+    pub fn new() -> MultipartRequest {
+        MultipartRequest {
+            boundary: format!("{}", uuid::Uuid::new_v4().simple()),
+            fields: Vec::new(),
+        }
+    }
```

Comment:
This a teeny nit pick, sorry. Since you'll be in this file again, the preferred style in reqwest is to have an empty line between methods.